stake-pool/program/src/processor.rs

```diff
@@ -0,0 +1,609 @@
+        let new_owner_info = next_account_info(account_info_iter)?;
+        let new_owner_fee_info = next_account_info(account_info_iter)?;
+
+        let mut stake_pool = State::deserialize(&stake_pool_info.data.borrow())?.stake_pool()?;
+
+        if *owner_info.key != stake_pool.owner {
+            return Err(Error::InvalidInput.into());
+        }
+        if !owner_info.is_signer {
+            return Err(Error::InvalidInput.into());
+        }
+        stake_pool.owner = *new_owner_info.key;
+        stake_pool.owner_fee_account = *new_owner_fee_info.key;
+        State::Init(stake_pool).serialize(&mut stake_pool_info.data.borrow_mut())?;
+        Ok(())
+    }
+    /// Processes an [Instruction](enum.Instruction.html).
+    pub fn process(program_id: &Pubkey, accounts: &[AccountInfo], input: &[u8]) -> ProgramResult {
+        let instruction = StakePoolInstruction::deserialize(input)?;
+        match instruction {
+            StakePoolInstruction::Initialize(init) => {
+                info!("Instruction: Init");
+                Self::process_initialize(program_id, init, accounts)
+            }
+            StakePoolInstruction::Deposit => {
+                info!("Instruction: Deposit");
+                Self::process_deposit(program_id, accounts)
+            }
+            StakePoolInstruction::Withdraw(amount) => {
+                info!("Instruction: Withdraw");
+                Self::process_withdraw(program_id, amount, accounts)
+            }
+            StakePoolInstruction::SetStakingAuthority => {
+                info!("Instruction: SetStakingAuthority");
+                Self::process_set_staking_auth(program_id, accounts)
+            }
+            StakePoolInstruction::SetOwner => {
+                info!("Instruction: SetOwner");
+                Self::process_set_owner(program_id, accounts)
+            }
+        }
+    }
+}
+
+// Test program id for the stake-pool program.
+#[cfg(not(target_arch = "bpf"))]
+const MY_PROGRAM_ID: Pubkey = Pubkey::new_from_array([2u8; 32]);
+
+/// Routes invokes to the token program, used for testing.
+#[cfg(not(target_arch = "bpf"))]
+pub fn invoke_signed<'a>(
+    instruction: &Instruction,
+    account_infos: &[AccountInfo<'a>],
+    signers_seeds: &[&[&[u8]]],
+) -> ProgramResult {
+    let mut new_account_infos = vec![];
+    for meta in instruction.accounts.iter() {
+        for account_info in account_infos.iter() {
+            if meta.pubkey == *account_info.key {
+                let mut new_account_info = account_info.clone();
+                for seeds in signers_seeds.iter() {
+                    let signer = Pubkey::create_program_address(seeds, &MY_PROGRAM_ID).unwrap();
+                    if *account_info.key == signer {
+                        new_account_info.is_signer = true;
+                    }
+                }
+                new_account_infos.push(new_account_info);
+            }
+        }
+    }
+    spl_token::processor::Processor::process(
```

Comment:
What's the plan to call the stake program in testing?  Is it waiting for https://github.com/solana-labs/solana/pull/12653 to land?